modules/spring-config/src/main/java/org/mule/runtime/config/internal/ModuleDelegatingEntityResolver.java

```diff
@@ -118,6 +117,29 @@ public InputSource resolveEntity(String publicId, String systemId) throws SAXExc
     return inputSource;
   }
 
+  private String overrideSystemIdForCompatibility(String publicId, String systemId) throws SAXException, IOException {
+    if (systemId.equals(CORE_XSD)) {
+      Boolean useDeprecated = muleEntityResolver.resolveEntity(publicId, CORE_DEPRECATED_XSD) != null;
+      Boolean usingCompatibility = muleEntityResolver.resolveEntity(publicId, COMPATIBILITY_XSD) != null;
+      Boolean runningTests = false;
+
+      try {
+        Class.forName("org.mule.tck.junit4.AbstractMuleTestCase", true, currentThread().getContextClassLoader());
```

Comment:
What happens is the application contains the mule-unit-test jar?

Reply:
In that case it would work.